unity-renderer/Assets/UIComponents/Scripts/Components/SectionSelector/SectionToggle.cs

```diff
@@ -41,20 +43,25 @@ public interface ISectionToggle
 public class SectionToggle : MonoBehaviour, ISectionToggle
 {
     [SerializeField] private Toggle toggle;
-    [SerializeField] private Image toggleBackground;
     [SerializeField] private TMP_Text sectionText;
     [SerializeField] private Image sectionImage;
-    [SerializeField] private Color selectedBackgroundColor;
+
+    [Header("Visual Configuration When Selected")]
+    [SerializeField] private ColorBlock backgroundTransitionColorsForSelected;
     [SerializeField] private Color selectedTextColor;
     [SerializeField] private Color selectedImageColor;
-    [SerializeField] private Color unselectedBackgroundColor;
+
+    [Header("Visual Configuration When Unselected")]
+    [SerializeField] private ColorBlock backgroundTransitionColorsForUnselected;
     [SerializeField] private Color unselectedTextColor;
     [SerializeField] private Color unselectedImageColor;
 
     public ToggleEvent onSelect => toggle?.onValueChanged;
 
     private void Awake() { ConfigureDefaultOnSelectAction(); }
 
+    private void OnEnable() { StartCoroutine(ForceToRefreshToggleState()); }
```

Comment:
We should use Coroutine Starter here!

Reply:
I disagree here.

`CoroutineStarter` has no benefit whatsoever besides allow you to start coroutines in a no Monobehaviour class. In this case this is just fine.

In fact using the `CoroutineStarter` has the downside of obscuring profiling. If you take a look at the profiler you will see that the `CoroutineStarter` is taking lot of time in the `MoveNext` operation of the `IEnumerators` and there's no way to know which ones are taking most of the time.

Reply:
Furthermore the `CoroutineStarter` would add you some tiresome boilerplate code. If your monobehaviour starts a coroutine, the lifecycle is attached to the monobehaviour. Disabling or removing a Monobehaviour stops and removes its running coroutines.

If you do it in a CoroutineStarter you might end up forgetting to stop the routine and it will produce a leak or some undesirable issues.